spec/v1.0-draft/digest_set.md

```diff
@@ -23,91 +23,130 @@ algorithms below use lowercase hex encoding. Usually there is just a
+designed for go modules but can be used to digest the _contents_ of an
+arbitrary archive or file tree. Equivalent to extracting the archive to an
+empty directory and running the following command in that directory:
+
+```bash
+find . -type f | cut -c3- | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
+```
+
+For example, the module dirhash
+`h1:Khu2En+0gcYPZ2kuIihfswbzxv/mIHXgzPZ018Oty48=` would be encoded as
+`{"dirHash1": "2a1bb6127fb481c60f67692e22285fb306f3c6ffe62075e0ccf674d7c3adcb8f"}`.
+
+<details>
+<summary>Detailed example</summary>
+
+The go module `github.com/marklodato/go-hello-world@v0.0.1` has module
+dirhash `h1:Khu2En+0gcYPZ2kuIihfswbzxv/mIHXgzPZ018Oty48=`:
+
+```bash
+$ curl https://sum.golang.org/lookup/github.com/marklodato/go-hello-world@v0.0.1
+...
+github.com/marklodato/go-hello-world v0.0.1 h1:Khu2En+0gcYPZ2kuIihfswbzxv/mIHXgzPZ018Oty48=
+...
+```
+
+To compute the dirhash by hand, first fetch the module archive and extract
+it to an empty directory:
+
+```bash
+curl -O https://proxy.golang.org/github.com/marklodato/go-hello-world/@v/v0.0.1.zip
+mkdir tmp
+cd tmp
+unzip ../v0.0.1.zip
+```
+
+We can see all of the files in the directory using the first part of the
+command above:
+
+```bash
+$ find . -type f | cut -c3- | LC_ALL=C sort | xargs -r sha256sum
+3a137eef6458bfb76bb2c63fc29ffc7166604d2d2e09ed9d8250a534122a8364  github.com/marklodato/go-hello-world@v0.0.1/README.md
+28e7c942a036902d981759d0bf5704d2bfc7cb500caf68b84711b234af01c6a5  github.com/marklodato/go-hello-world@v0.0.1/go.mod
+ddc4da627d9a9f45fb29641a1b185d6f53287ecfd921aacbf4fe54b7a86fe8d1  github.com/marklodato/go-hello-world@v0.0.1/main.go
+```
+
+The dirhash is the sha256 sum over the output of the previous command:
+
+```bash
+$ find . -type f | cut -c3- | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
+2a1bb6127fb481c60f67692e22285fb306f3c6ffe62075e0ccf674d7c3adcb8f
+```
+
+This is equivalent to the base64 encoded version:
+
+```bash
+$ echo '2a1bb6127fb481c60f67692e22285fb306f3c6ffe62075e0ccf674d7c3adcb8f' | xxd -r -p | {printf 'h1:'; base64}
+h1:Khu2En+0gcYPZ2kuIihfswbzxv/mIHXgzPZ018Oty48=
+```
+
+</details>
+
+#### `gitCommit`, `gitTree`, `gitBlob`, `gitTag`
+
+The lowercase hex SHA-1 (40 character) or SHA-256 (64 character) of a git
+commit, tree, blob, or tag object, respectively. The `gitTree` and `gitBlob` in
+particular can be used for arbitrary trees or files, even outside git.
+
+This hash is computed over `<type> SP <size> NUL <content>`, where:
+
+-   `<type>` is one of `commit`, `tree`, `blob`, `tag`
+-   `SP` is the ASCII space character, 0x20
+-   `<size>` is the number of bytes in `<content>`, represented as a decimal
+    ASCII number with no leading zeros
+-   `NUL` is the ASCII NUL character, 0x00
+-   `<content>` is git representation of the object:
+    -   For `commit`, the raw commit object
+        ([reference](https://stackoverflow.com/a/37438460/303425))
```

Comment:
+1

Reply:
As I said, there is no better reference that I can find. I spent hours looking. Please let me know if you find something else.

In the meantime, since everyone has asked this question, I added a footnote explaining.